application/src/main/java/org/togetherjava/tjbot/commands/utils/Arraylizable.java

```diff
@@ -0,0 +1,16 @@
+package org.togetherjava.tjbot.commands.utils;
+
+/**
+ * An interface with just a {@link Arraylizable#toArray()} method. This is helpful for passing
+ * classes that provide component arguments.
+ *
+ * @param <T> The type of the array. Usually {@link String} for component arguments
+ */
+public interface Arraylizable<T> {
```

Comment:
before i dive deeper into the code review, let me just say right now that im against adding such a general purpose interface.

i dont see this to become something that is used in more packages, no need to introduce an interface. and the name of the interface is also a bit odd.

Reply:
I see this being useful, like the javadoc sais, for passing a class holding component arguments. I chose the name after the Serializable class but i would happily recieve better naming suggestions

Reply:
its just that too generic solutions are rarely appropriate, unless they are actually used multiple times.

for the same reason i replaced illus "lookaheadqueue" and "skipablelookaheadqueue" classes (with their two interfaces and 2 classes and abstract classes and ...) by just a single class that was specialized for the task at hand `TokenQueue`.

similar for interfaces, they only really start making sense once u have more than one implementation (or can forsee multiple implementations in the very near future).
in this case it feels more like u created an interface that will likely always just have your implementation and nothing more.

Reply:
When creating this interface i had all classes implementing UserInteractor in mind which is basically any feature. The use case for this interface is small but when having multiple argument holding classes this makes it a lot easier to generate the component IDs as you do not have to call toArray on every class before passing the array to generateComponentId but can pass anything arraylizable to generateComponentId

Reply:
If we keep this class we should probably make ComponentIdInteractor.generateComponentId accept Arraylizable

Reply:
the method is supposed to take varargs, not direct arrays. `generateComponentIds(foo, bar, baz)`. is there an actual use case for giving it an actual array somewhere?

Reply:
You are right it is not directly supposed to be used with arrays although it is possible. The interface could be renamed to `ComponentArgsHolder` with a `getStringArgs` method. That would possibly make things more clear.

Reply:
i mean, im not against using the method with an array, if its better for the user in that case. im just asking whether u had a need for that here. maybe u just didnt notice that its varargs (i didnt check ur code yet)

Reply:
I did notice that. I just did not want to call toArray in my args classes every time manually when creating the component IDs

Reply:
Wait I just realized that RemoveComponentArguments extends ComponentArguments anyways. I don't even need the interface anymore. I think I wanted to use that because I tried using records first. That did not work though as they are final

Reply:
Removed with rework